app/src/main/java/com/hedvig/app/feature/marketpicker/LanguageAdapter.kt

```diff
@@ -16,74 +13,69 @@ import com.hedvig.app.util.spring
 import kotlinx.android.synthetic.main.language_item_new.view.*
 
 class LanguageAdapterNew(
-    private val languageAndMarketViewModel: LanguageAndMarketViewModel,
+    private val model: LanguageAndMarketViewModel,
     private val selectedMarket: Market
 ) : RecyclerView.Adapter<LanguageAdapterNew.ViewHolder>() {
-    private var lastChecked: RadioButton? = null
-    private var lastCheckedPos = 0
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ViewHolder(
-        LayoutInflater.from(parent.context)
-            .inflate(R.layout.language_item_new, parent, false)
-    )
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = ViewHolder(parent)
 
     override fun getItemCount() = Market.values().size
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        languageAndMarketViewModel.selectedLanguage.postValue(null)
+        model.selectedLanguage.postValue(null)
```

Comment:
Re-binding can happen under many circumstances in the RecyclerView, should this side-effect be in `onBindViewHolder()`?